libs/sql-connection/src/lib.rs

```diff
@@ -0,0 +1,410 @@
+    fn query_raw(&self, db: &str, sql: &str, params: &[ParameterizedValue<'_>]) -> Result<ResultSet, QueryError>;
+
+    /// See
+    /// https://prisma.github.io/prisma-query/prisma_query/connector/trait.Queryable.html#tymethod.execute_raw
+    ///
+    /// The `db` param is only used on SQLite to give a name to the attached database.
+    fn execute_raw(&self, db: &str, sql: &str, params: &[ParameterizedValue<'_>]) -> Result<u64, QueryError>;
+}
+
+/// A generic asynchronous SQL connection interface.
+#[async_trait::async_trait]
+pub trait SqlConnection: Send + Sync + 'static {
+    /// See
+    /// https://prisma.github.io/prisma-query/prisma_query/connector/trait.Queryable.html#tymethod.execute
+    ///
+    /// The `db` param is only used on SQLite to give a name to the attached database.
+    async fn execute<'a>(&self, db: &str, q: Query<'a>) -> Result<Option<Id>, QueryError>;
+
+    /// See
+    /// https://prisma.github.io/prisma-query/prisma_query/connector/trait.Queryable.html#tymethod.query
+    ///
+    /// The `db` param is only used on SQLite to give a name to the attached database.
+    async fn query<'a>(&self, db: &str, q: Query<'a>) -> Result<ResultSet, QueryError>;
+
+    /// See
+    /// https://prisma.github.io/prisma-query/prisma_query/connector/trait.Queryable.html#tymethod.query_raw
+    ///
+    /// The `db` param is only used on SQLite to give a name to the attached database.
+    async fn query_raw<'a>(
+        &self,
+        db: &str,
+        sql: &str,
+        params: &[ParameterizedValue<'a>],
+    ) -> Result<ResultSet, QueryError>;
+
+    /// See
+    /// https://prisma.github.io/prisma-query/prisma_query/connector/trait.Queryable.html#tymethod.execute_raw
+    ///
+    /// The `db` param is only used on SQLite to give a name to the attached database.
+    async fn execute_raw<'a>(&self, db: &str, sql: &str, params: &[ParameterizedValue<'a>]) -> Result<u64, QueryError>;
+}
+
+type SqlitePool = Pool<SqliteManager>;
+
+/// A pooled of connections to an SQLite database. It exposes both sync and async query
+/// interfaces.
+pub struct Sqlite {
+    pool: SqlitePool,
+    file_path: String,
+    // TODO: remove this when we remove the sync API
+    runtime: Runtime,
+}
+
+impl Sqlite {
+    /// Create a connection pool to an SQLite database.
+    pub fn new(url: &str) -> Result<Self, QueryError> {
+        let params = SqliteParams::try_from(url)?;
+        let file_path = params.file_path.to_str().unwrap().to_string();
+
+        let pool = prisma_query::pool::sqlite(url)?;
+
+        Ok(Self {
+            pool,
+            file_path,
+            runtime: default_runtime(),
+        })
+    }
+
+    /// The filesystem path of connection's database.
+    pub fn file_path(&self) -> &str {
+        self.file_path.as_str()
+    }
+
+    async fn get_connection(&self, db: &str) -> Result<CheckOut<SqliteManager>, QueryError> {
+        let conn = self.pool.check_out().await?;
+
+        conn.execute_raw(
```

Comment:
Maybe this could happen once in the initialization? We could skip the dbname param...

Reply:
I thought about doing it, but I wanted to finish porting the existing API first. I can try to do that tomorrow morning.

btw if we do that, do you think there is still a value to the `SqlConnection` trait (the async one), or should we implement `Queryable` directly?

Reply:
What extra you'd bring over queryable? This interface was made to hide the original &mut selt in queryable. Now it's not needed and I don't see so much need for that mutex either...

Reply:
Indeed! I forgot Queryable doesn't need mutable references anymore. I'll try removing the `db` parameter and the mutexes tomorrow.